Automattic/jetpack
# Contact Form: revalidate fields on blur

Fixes https://github.com/Automattic/jetpack/issues/34281

## Proposed changes:

https://github.com/Automattic/jetpack/pull/34173 made the form validation more accessible. This PR iterates on it by revalidating the invalid fields on blur (i.e., focus out) so that users receive quicker feedback. We chose not to revalidate on input change because this behavior is confusing/irritating for most inputs.

### Other information:


## Jetpack product discussion


Project thread: pf5801-87-p2

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






Review in conjunction with https://github.com/Automattic/jetpack/pull/34400.

- Spin up a test site with this branch
- Create a new post
- Add a _Contact form_ block to the post
- Add at least a text, Multiple Choice, and Single Choice fields
- Make them required
- View the preview
- Check that when one or more fields are invalid, an error message is displayed above the submit button and below each invalid field
- Fix the error of each field and notice the error message disappears when focusing out
- Make sure you can submit the form successfully
